instrumentation/pulsar/pulsar-2.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarClientSuppressReceiveSpansTest.java

```diff
@@ -0,0 +1,388 @@
+                span ->
+                    span.hasName(topic + " receive")
+                        .hasKind(SpanKind.CONSUMER)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            receiveAttributes(topic, msgId.toString(), false))));
+  }
+
+  @Test
+  void testConsumeNonPartitionedTopicUsingReceiveAsync() throws Exception {
+    String topic = "persistent://public/default/testConsumeNonPartitionedTopicCallReceiveAsync";
+    admin.topics().createNonPartitionedTopic(topic);
+    consumer =
+        client
+            .newConsumer(Schema.STRING)
+            .subscriptionName("test_sub")
+            .topic(topic)
+            .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+            .subscribe();
+
+    producer = client.newProducer(Schema.STRING).topic(topic).enableBatching(false).create();
+
+    CompletableFuture<Message<String>> result =
+        consumer
+            .receiveAsync()
+            .whenComplete(
+                (message, throwable) -> {
+                  if (message != null) {
+                    testing.runWithSpan("callback", () -> acknowledgeMessage(consumer, message));
+                  }
+                });
+
+    String msg = "test";
+    MessageId msgId = testing.runWithSpan("parent", () -> producer.send(msg));
+
+    result.get(1, TimeUnit.MINUTES);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName(topic + " publish")
+                        .hasKind(SpanKind.PRODUCER)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            sendAttributes(topic, msgId.toString(), false)),
+                span ->
+                    span.hasName(topic + " receive")
+                        .hasKind(SpanKind.CONSUMER)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            receiveAttributes(topic, msgId.toString(), false)),
+                span ->
+                    span.hasName("callback")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(2))));
+  }
+
+  @Test
+  void testConsumeNonPartitionedTopicUsingReceiveWithTimeout() throws Exception {
+    String topic =
+        "persistent://public/default/testConsumeNonPartitionedTopicCallReceiveWithTimeout";
+    admin.topics().createNonPartitionedTopic(topic);
+    consumer =
+        client
+            .newConsumer(Schema.STRING)
+            .subscriptionName("test_sub")
+            .topic(topic)
+            .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+            .subscribe();
```

Comment:
Wait, where does the "receive with timeout" part factor into this test? I'm not seeing it. If it's there, perhaps a comment can help to clarify, because a reader doesn't immediately see how the receive-with-timeout is wired up.

Reply:
I presume it is `consumer.receive(1, TimeUnit.MINUTES);`